extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -642,11 +669,23 @@ export class AmpAutocomplete extends AMP.BaseElement {
     if (delta === 0 || !this.resultsShowing_()) {
       return Promise.resolve();
     }
+    // Active element logic
     const keyUpWhenNoneActive = this.activeIndex_ === -1 && delta < 0;
     const index = keyUpWhenNoneActive ? delta : this.activeIndex_ + delta;
     const activeIndex = mod(index, this.container_.children.length);
     const newActiveElement = this.container_.children[activeIndex];
     this.inputElement_.value = newActiveElement.getAttribute('data-value');
+
+    // Element visibility logic
+    const {offsetTop: itemTop, offsetHeight: itemHeight} = newActiveElement;
```

Comment:
These read/writes ([scrollTop, offsetHeight, offsetTop](https://gist.github.com/paulirish/5d52fb081b3570c81e3a)) need to be in a measureMutate block since they cause relayout/reflow

Reply:
Thanks for this reminder 😄